text_extensions_for_pandas/array/span.py

```diff
@@ -878,6 +1078,33 @@ def repr_html_show_offsets(self):
     def repr_html_show_offsets(self, show_offsets: bool):
         self._repr_html_show_offsets = show_offsets
 
+    ##########################################
+    # Keep private and protected methods here.
+
+    @memoized_property
+    def _hash(self):
+        return hash((self.target_text.tobytes(), self._begins.tobytes(),
+                     self._ends.tobytes()))
+
+    # noinspection PyMethodMayBeStatic
+    def _cached_property_names(self) -> List[str]:
+        """
+        :return: names of cached properties whose values are computed on demand
+         and invalidated when the set of spans change.
+        """
+        return ["_hash", "is_single_document", "target_text",
+                "normalized_covered_text", "document_text"]
+
+    def _clear_cached_properties(self) -> None:
+        """
+        Remove cached values of memoized properties to reflect changes to the
+        data on which they are based.
+        """
+        for n in self._cached_property_names():
+            attr_name = "_{0}".format(n)
+            if hasattr(self, attr_name):
+                delattr(self, attr_name)
```

Comment:
Hmm, this kind of sucks you have to do this to clear it.  So you remove the attr, does it need to be added back?

Reply:
The `@memoized_property` decorator will add the field back the next time the property is accessed. See https://github.com/estebistec/python-memoized-property/blob/17424f077037a361cd09f03e4e6376e76361281c/memoized_property.py#L7

Yes, it's not pretty. For what it's worth, the new `@functools.cached_property` decorator in Python 3.8 is worse. It stores the cached value in a field with the same name as the property.